tests/unit/plugins/modules/notification/test_slack.py

```diff
@@ -88,6 +88,39 @@ def test_message_with_thread(self):
             assert call_data['thread_ts'] == '100.00'
             assert fetch_url_mock.call_args[1]['url'] == "https://hooks.slack.com/services/XXXX/YYYY/ZZZZ"
 
+    # https://github.com/ansible-collections/community.general/issues/1097
+    def test_ts_in_message_does_not_cause_edit(self):
+        set_module_args({
+            'token': 'xoxa-123456789abcdef',
+            'msg': 'test with ts'
+        })
+
+        with patch.object(slack, "fetch_url") as fetch_url_mock:
+            fetch_url_mock.return_value = (None, {"status": 200})
```

Comment:
The module code assumes that the `None` you return here as a `read()` method which returns a JSON encoded as a string.